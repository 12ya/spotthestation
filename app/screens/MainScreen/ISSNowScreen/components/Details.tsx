import React, { useEffect, useState } from "react"
import { TextStyle, View, ViewStyle } from "react-native"
import { Text, Toggle } from "../../../../components"
import { colors, fontSizes, scale, spacing } from "../../../../theme"
import { convertToImperial } from "../../../../utils/convertToImperial"
import { formatDate } from "../../../../utils/formatDate"

const locationDetails = {
  latitude: 100,
  longitude: 100,
  altitude: 400,
  speed: 27000,
}

export function Details() {
  const [isImperial, setIsImperial] = useState(false)
  const [currentDateTime, setCurrentDateTime] = useState(new Date().toISOString())

  useEffect(() => {
    const secTimer = setInterval(() => {
      setCurrentDateTime(new Date().toISOString())
    }, 1000)

    return () => clearInterval(secTimer)
  }, [])

  return (
    <View style={$container}>
      <View style={[$rowContainer, $centered]}>
        <Text text={`${formatDate(currentDateTime, "dd MMM yyyy, k:mm:ss")} (EST)`} style={$text} />
      </View>
      <View style={$rowContainer}>
        <View style={$columnContainer}>
          <Text tx="issNowTab.details.latitude" style={$label} />
          <Text text={`${locationDetails.latitude} S`} style={$text} />
        </View>
        <View style={$columnContainer}>
          <Text tx="issNowTab.details.longitude" style={$label} />
          <Text text={`${locationDetails.longitude} W`} style={$text} />
        </View>
        <View style={$columnContainer}>
          <Text tx="issNowTab.details.altitude" style={$label} />
          <Text
            text={`${
              isImperial
                ? `${convertToImperial(locationDetails.altitude, "distance")} mi`
                : `${locationDetails.altitude} km"`
            }`}
            style={$text}
          />
        </View>
        <View style={$columnContainer}>
          <Text tx="issNowTab.details.speed" style={$label} />
          <Text
            text={`${
              isImperial
                ? `${convertToImperial(locationDetails.speed, "speed")} mi/h`
                : `${locationDetails.speed} km/h`
            }`}
            style={$text}
          />
        </View>
      </View>
      <View style={[$rowContainer, $centered]}>
        <Text tx="issNowTab.details.switchLabel" style={[$text, $margin]} />
        <Toggle variant="switch" onValueChange={setIsImperial} value={isImperial} />
      </View>
    </View>
  )
}

const $container: ViewStyle = {
  display: "flex",
  width: "100%",
  height: "23%",
  backgroundColor: colors.backgroundDark,
  padding: scale(spacing.medium),
}

const $centered: ViewStyle = {
  justifyContent: "center",
}

const $margin: ViewStyle = {
  marginRight: scale(spacing.extraSmall),
}

const $rowContainer: ViewStyle = {
  display: "flex",
  flexDirection: "row",
  alignItems: "center",
  justifyContent: "space-between",
  marginBottom: scale(spacing.medium),
}

const $columnContainer: ViewStyle = {
  display: "flex",
  alignItems: "flex-start",
}

const $text: TextStyle = {
  fontSize: fontSizes[18],
  color: colors.palette.neutral100,
}

const $label: TextStyle = {
  fontSize: fontSizes[16],
  fontWeight: "bold",
  color: colors.palette.neutral100,
}
