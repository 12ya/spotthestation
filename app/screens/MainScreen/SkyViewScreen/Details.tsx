import React from "react"
import { ViewStyle, View, PressableProps, TextStyle } from "react-native"
import { Icon, Text } from "../../../components"
import { OrbitPoint } from "../../../services/api/api.types"
import { fontSizes, lineHeights, scale, typography } from "../../../theme"
import { colors } from "../../../theme/colors"
import { calculateDistance, calculateOrbitalSpeed } from "../components/helpers"

export interface DetailsProps {
  /**
   * A function for closing modal.
   */
  onClose?: PressableProps["onPress"],
  issData: OrbitPoint
  observer: [number, number]
}

export function Details({ onClose, issData, observer }: DetailsProps) {
  const distance = (data: OrbitPoint): number => {
    return Math.round(calculateDistance(observer[0], observer[1], 0, data.latitude, data.longitude, data.elevation)/1000)
  }
  return (
    <View style={$modalBodyContainer}>
      <Icon icon="x" 
        accessible
        accessibilityLabel="x button"
        accessibilityHint="close modal"
        accessibilityRole="button"
        color={colors.palette.neutral450} 
        onPress={onClose} 
        containerStyle={$close} 
        size={36}
      />
      <View style={$contentContainer}>
        <Text
          accessible
          accessibilityLabel="modal title"
          accessibilityHint="modal title"
          accessibilityRole="text"
          tx="issView.details.title" 
          style={$title}
        />
        <View style={$buttonsContainer}>
          <View 
            accessible
            accessibilityLabel="distance"
            accessibilityHint="distance"
            accessibilityRole="text"
            style={$detailBox}
          >
            <Text tx="issView.details.distance" style={$detailTitle} />
            <Text text={`${distance(issData)} km`} style={$detailValue} />
          </View>
          <View 
            accessible
            accessibilityLabel="orbital Speed"
            accessibilityHint="orbital Speed"
            accessibilityRole="text"
            style={$detailBox}
          >
            <Text tx="issView.details.orbitalSpeed" style={$detailTitle} />
            <Text text={`${calculateOrbitalSpeed(issData.latitude, issData.azimuth, issData.elevation)} M/S`} style={$detailValue} />
          </View>
          <View 
            accessible
            accessibilityLabel="longitude"
            accessibilityHint="longitude"
            accessibilityRole="text"
            style={$detailBox}
          >
            <Text tx="issView.details.longitude" style={$detailTitle} />
            <Text text={issData.longitude.toFixed(2)} style={$detailValue} />
          </View>
          <View 
            accessible
            accessibilityLabel="latitude"
            accessibilityHint="latitude"
            accessibilityRole="text"
            style={$detailBox}
          >
            <Text tx="issView.details.latitude" style={$detailTitle} />
            <Text text={issData.latitude.toFixed(2)} style={$detailValue} />
          </View>
          <View 
            accessible
            accessibilityLabel="altitude"
            accessibilityHint="altitude"
            accessibilityRole="text"
            style={$detailBox}
          >
            <Text tx="issView.details.altitude" style={$detailTitle} />
            <Text text={issData.elevation.toFixed(2)} style={$detailValue} />
          </View>
          <View 
            accessible
            accessibilityLabel="crew On board"
            accessibilityHint="crew On board"
            accessibilityRole="text"
            style={$detailBox}
          >
            <Text tx="issView.details.crewOnboard" style={$detailTitle} />
            <Text text="7" style={$detailValue} />
          </View>
        </View>
        <View style={$buttonsContainer}>
          <View 
            accessible
            accessibilityLabel="launched"
            accessibilityHint="launched"
            accessibilityRole="text"
            style={$detailRow}
          >
            <Text tx="issView.details.launched" style={$detailRowTitle} />
            <Text text={`20 November 1998 (${(new Date().getFullYear()) - (new Date('1998-11-20').getFullYear())} yrs ago)`} style={$detailRowValue} />
          </View>
          <View 
            accessible
            accessibilityLabel="mass"
            accessibilityHint="mass"
            accessibilityRole="text"
            style={$detailRow}
          >
            <Text tx="issView.details.mass" style={$detailRowTitle} />
            <Text text="444,615 kg" style={$detailRowValue} />
          </View>
          <View 
            accessible
            accessibilityLabel="length"
            accessibilityHint="length"
            accessibilityRole="text"
            style={$detailRow}
          >
            <Text tx="issView.details.length" style={$detailRowTitle} />
            <Text text="73.0 m" style={$detailRowValue} />
          </View>
          <View 
            accessible
            accessibilityLabel="orbital Period"
            accessibilityHint="orbital Period"
            accessibilityRole="text"
            style={$detailRow}
          >
            <Text tx="issView.details.orbitalPeriod" style={$detailRowTitle} />
            <Text text="92.9 min" style={$detailRowValue} />
          </View>
          <View 
            accessible
            accessibilityLabel="orbits Per Day"
            accessibilityHint="orbits Per Day"
            accessibilityRole="text"
            style={$detailRow}
          >
            <Text tx="issView.details.orbitsPerDay" style={$detailRowTitle} />
            <Text text="15.49" style={$detailRowValue} />
          </View>
          <View 
            accessible
            accessibilityLabel="launched"
            accessibilityHint="launched"
            accessibilityRole="text"
            style={$detailRow}
          >
            <Text tx="issView.details.launched" style={$detailRowTitle} />
            <Text text="2 km / month" style={$detailRowValue} />
          </View>
          <View 
            accessible
            accessibilityLabel="launched"
            accessibilityHint="launched"
            accessibilityRole="text"
            style={$detailRow}
          >
            <Text tx="issView.details.launched" style={$detailRowTitle} />
            <Text text={`133,312\n(as of June 2022)`} style={$detailRowValue} />
          </View>
        </View>
      </View>
    </View>
  )
}

const $modalBodyContainer: ViewStyle = {
  backgroundColor: colors.palette.neutral350,
  borderTopLeftRadius: scale(18),
  borderTopRightRadius: scale(18),
}

const $contentContainer: ViewStyle = {
  width: "100%",
  paddingHorizontal: scale(36),
  paddingBottom: scale(52)
}

const $buttonsContainer: ViewStyle = {
  width: "100%",
  flexDirection: "row",
  flexWrap: "wrap",
  justifyContent: "space-between"
}

const $close: ViewStyle = {
  position: "absolute",
  top: 0,
  right: 0,
  padding: scale(18),
  zIndex: 5
}

const $title: TextStyle = {
  color: colors.palette.neutral250,
  width: "95%",
  fontSize: fontSizes[32],
  fontFamily: typography.primary.normal,
  lineHeight: lineHeights[44],
  paddingBottom: scale(36),
  marginTop: scale(36)
}

const $detailBox: ViewStyle = {
  backgroundColor: colors.palette.overlayWhite,
  width: "48%",
  borderRadius: scale(10),
  padding: scale(8),
  marginBottom: scale(18)
}

const $detailTitle: TextStyle = {
  fontFamily: typography.primary.normal,
  color: colors.palette.neutral450,
  fontSize: fontSizes[13],
  lineHeight: lineHeights[16],
  textAlign: "center",
  textTransform: "uppercase"
}

const $detailValue: TextStyle = {
  fontFamily: typography.primary.normal,
  color: colors.palette.neutral250,
  fontSize: fontSizes[22],
  lineHeight: lineHeights[28],
  textAlign: "center",
}

const $detailRow: ViewStyle = {
  width: "100%",
  marginBottom: scale(18),
  flexDirection: "row",
  justifyContent: "space-between"
}

const $detailRowTitle: TextStyle = {
  fontFamily: typography.primary.normal,
  color: colors.palette.neutral450,
  fontSize: fontSizes[18],
  lineHeight: lineHeights[21],
  textAlign: "left",
}

const $detailRowValue: TextStyle = {
  fontFamily: typography.primary.normal,
  color: colors.palette.neutral100,
  fontSize: fontSizes[18],
  lineHeight: lineHeights[21],
  textAlign: "right",
  maxWidth: scale(155)
}
