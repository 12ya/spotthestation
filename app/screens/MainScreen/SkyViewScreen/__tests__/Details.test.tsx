/* eslint-disable react/display-name */
/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
import React from "react"
import renderer from 'react-test-renderer'
import { Details } from '../Details'

it('renders correctly', () => {
  const tree = renderer
    .create(<Details 
              issData={{ date: "12-12-2012", latitude: 0, longitude: 0, azimuth: 0, elevation: 0, altitude: 0 }}
              observer={[0,0]}
            />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
