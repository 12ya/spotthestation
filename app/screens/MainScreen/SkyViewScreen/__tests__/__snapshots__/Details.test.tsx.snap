// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1E2022",
      "borderTopLeftRadius": 26,
      "borderTopRightRadius": 26,
    }
  }
>
  <View
    style={
      Object {
        "paddingBottom": 75,
        "paddingHorizontal": 52,
        "width": "100%",
      }
    }
  >
    <Text
      accessibilityHint="modal title"
      accessibilityLabel="modal title"
      accessibilityRole="text"
      accessible={true}
      allowFontScaling={false}
      style={
        Array [
          Object {},
          Array [
            Object {
              "fontSize": 23,
              "lineHeight": 35,
            },
            Object {
              "fontFamily": "Helvetica Neue",
            },
            Object {
              "color": "#191015",
            },
          ],
          undefined,
          undefined,
          Object {
            "color": "#DEDDDE",
            "fontFamily": "Helvetica Neue",
            "fontSize": 46,
            "lineHeight": 63,
            "marginTop": 52,
            "paddingBottom": 52,
            "width": "95%",
          },
        ]
      }
    >
      issView.details.title undefined
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityHint="distance"
        accessibilityLabel="distance"
        accessibilityRole="text"
        accessible={true}
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.2)",
            "borderRadius": 15,
            "marginBottom": 26,
            "padding": 12,
            "width": "48%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#ADADAE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 19,
                "lineHeight": 23,
                "textAlign": "center",
                "textTransform": "uppercase",
              },
            ]
          }
        >
          issView.details.distance undefined
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 32,
                "lineHeight": 40,
                "textAlign": "center",
              },
            ]
          }
        >
          0 km
        </Text>
      </View>
      <View
        accessibilityHint="orbital Speed"
        accessibilityLabel="orbital Speed"
        accessibilityRole="text"
        accessible={true}
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.2)",
            "borderRadius": 15,
            "marginBottom": 26,
            "padding": 12,
            "width": "48%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#ADADAE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 19,
                "lineHeight": 23,
                "textAlign": "center",
                "textTransform": "uppercase",
              },
            ]
          }
        >
          issView.details.orbitalSpeed undefined
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 32,
                "lineHeight": 40,
                "textAlign": "center",
              },
            ]
          }
        >
          8374 M/S
        </Text>
      </View>
      <View
        accessibilityHint="longitude"
        accessibilityLabel="longitude"
        accessibilityRole="text"
        accessible={true}
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.2)",
            "borderRadius": 15,
            "marginBottom": 26,
            "padding": 12,
            "width": "48%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#ADADAE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 19,
                "lineHeight": 23,
                "textAlign": "center",
                "textTransform": "uppercase",
              },
            ]
          }
        >
          issView.details.longitude undefined
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 32,
                "lineHeight": 40,
                "textAlign": "center",
              },
            ]
          }
        >
          0
        </Text>
      </View>
      <View
        accessibilityHint="latitude"
        accessibilityLabel="latitude"
        accessibilityRole="text"
        accessible={true}
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.2)",
            "borderRadius": 15,
            "marginBottom": 26,
            "padding": 12,
            "width": "48%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#ADADAE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 19,
                "lineHeight": 23,
                "textAlign": "center",
                "textTransform": "uppercase",
              },
            ]
          }
        >
          issView.details.latitude undefined
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 32,
                "lineHeight": 40,
                "textAlign": "center",
              },
            ]
          }
        >
          0
        </Text>
      </View>
      <View
        accessibilityHint="altitude"
        accessibilityLabel="altitude"
        accessibilityRole="text"
        accessible={true}
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.2)",
            "borderRadius": 15,
            "marginBottom": 26,
            "padding": 12,
            "width": "48%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#ADADAE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 19,
                "lineHeight": 23,
                "textAlign": "center",
                "textTransform": "uppercase",
              },
            ]
          }
        >
          issView.details.altitude undefined
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 32,
                "lineHeight": 40,
                "textAlign": "center",
              },
            ]
          }
        >
          0 km
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityHint="Assembly Began"
        accessibilityLabel="Assembly Began"
        accessibilityRole="text"
        accessible={true}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 26,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#ADADAE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "lineHeight": 30,
                "textAlign": "left",
              },
            ]
          }
        >
          issView.details.launched undefined
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "lineHeight": 30,
                "maxWidth": 222,
                "textAlign": "right",
              },
            ]
          }
        >
          20 November 1998
        </Text>
      </View>
      <View
        accessibilityHint="crew On board"
        accessibilityLabel="crew On board"
        accessibilityRole="text"
        accessible={true}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 26,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#ADADAE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "lineHeight": 30,
                "textAlign": "left",
              },
            ]
          }
        >
          issView.details.crewOnboard undefined
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "lineHeight": 30,
                "maxWidth": 222,
                "textAlign": "right",
              },
            ]
          }
        >
          7
        </Text>
      </View>
      <View
        accessibilityHint="Estimated mass"
        accessibilityLabel="Estimated mass"
        accessibilityRole="text"
        accessible={true}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 26,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#ADADAE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "lineHeight": 30,
                "textAlign": "left",
              },
            ]
          }
        >
          issView.details.mass undefined
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "lineHeight": 30,
                "maxWidth": 222,
                "textAlign": "right",
              },
            ]
          }
        >
          462,000 kg
        </Text>
      </View>
      <View
        accessibilityHint="Estimated dimensions"
        accessibilityLabel="Estimated dimensions"
        accessibilityRole="text"
        accessible={true}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 26,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#ADADAE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "lineHeight": 30,
                "textAlign": "left",
              },
            ]
          }
        >
          issView.details.dimensions undefined
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "lineHeight": 30,
                "maxWidth": 222,
                "textAlign": "right",
              },
            ]
          }
        >
          109m wide x 73m long x 14m tall
        </Text>
      </View>
      <View
        accessibilityHint="orbital Period"
        accessibilityLabel="orbital Period"
        accessibilityRole="text"
        accessible={true}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 26,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#ADADAE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "lineHeight": 30,
                "textAlign": "left",
              },
            ]
          }
        >
          issView.details.orbitalPeriod undefined
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "lineHeight": 30,
                "maxWidth": 222,
                "textAlign": "right",
              },
            ]
          }
        >
          92.9 min
        </Text>
      </View>
      <View
        accessibilityHint="orbits Per Day"
        accessibilityLabel="orbits Per Day"
        accessibilityRole="text"
        accessible={true}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 26,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#ADADAE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "lineHeight": 30,
                "textAlign": "left",
              },
            ]
          }
        >
          issView.details.orbitsPerDay undefined
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "lineHeight": 30,
                "maxWidth": 222,
                "textAlign": "right",
              },
            ]
          }
        >
          15.49
        </Text>
      </View>
      <View
        accessibilityHint="Orbital Decay"
        accessibilityLabel="Orbital Decay"
        accessibilityRole="text"
        accessible={true}
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 26,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#ADADAE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "lineHeight": 30,
                "textAlign": "left",
              },
            ]
          }
        >
          issView.details.orbitalDecay undefined
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {},
              Array [
                Object {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                Object {
                  "fontFamily": "Helvetica Neue",
                },
                Object {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Helvetica Neue",
                "fontSize": 26,
                "lineHeight": 30,
                "maxWidth": 222,
                "textAlign": "right",
              },
            ]
          }
        >
          2 km / month
        </Text>
      </View>
    </View>
  </View>
</View>
`;
